### ISSUE TITLE
json encode (and decode) failing silently

### ISSUE THREAD

**Issue by linaori** [thirtybees/postnl-api-php#65]
I was facing an issue where I encountered "HTTP/1.1 500 Internal Server Error" errors on the PostNL side. Turns out that due to wrong character encoding, json_encode was failing silently, causing an empty body to be sent. 

I did a quick check and `JSON_THROW_ON_ERROR` isn't used so far. I don't think there should be any internal error handling for json errors, I think the handling of this is the responsibility of the Application using the library. My previous issue #58 was in fact caused by this as well.

There seems to be no explicit error handling for status 500. The response object (when `Unable to generate label` has:
```
reasonPhrase = 'Internal Server Error'
statusCode => 500
$response->getBody()->getContents() => ''
```

The logic in the LabelingService:
```php
if (200 === $response->getStatusCode()) {
    throw new ResponseException('Invalid API response', null, null, $response);
}

throw new NotFoundException('Unable to generate label');
```

Removing the status code check and then just always throwing this would be sufficient if there's no specific edge cases you want to cover here

```php
throw new ResponseException('Invalid API response', null, null, $response);
```

I can make PR with these changes if you want

**Comment by firstred** [comment]
Thanks for raising this issue! This was unfortunately not possible with the v1.x.x range of this library, because it had to support older versions of PHP.
I'll try to make the suggested changes in v2.0.0.

**Comment by firstred** [comment]
Hmmz, reading this issue again, I just realized my earlier reply makes no sense. Apologies for that!
There's absolutely no reason not to make this change in v1.x.x. It's not a matter of compatibility, except for the `JSON_THROW_ON_ERROR` flag, since not all PHP versions suported by v1.x.x have this flag.

Version 2.0.0 currently ignores the status code as well, and that works fine in production: https://github.com/firstred/postnl-api-php/blob/30034d902bd61b0d374f778b309a4a11c2db6c2e/src/Service/ResponseProcessor/Rest/LabellingServiceRestResponseProcessor.php#L67-L77
https://github.com/firstred/postnl-api-php/blob/30034d902bd61b0d374f778b309a4a11c2db6c2e/src/Service/ResponseProcessor/Rest/AbstractRestResponseProcessor.php#L55-L136

I don't think I have documented it properly, but these were supposed to be the meaning of every exception:
- `ResponseException`: Something wrong with the server response
- `ApiException`: Something wrong with the way the API returns an object (likely an unsupported feature or structure)
- `CifDownException`: CIF API broken
- `CifException`: Exception thrown by the CIF API
- `NotFoundException`: requested object cannot be found (!= cannot be generated)

The `NotFoundException` is really out of place here, tbh. I'll try to make your suggested changes in 1.4.0. Thanks again!
